src/spi/clone/CachingDirectoryManager.ts

```diff
@@ -60,13 +67,116 @@ export const CachingDirectoryManager: DirectoryManager = {
                 }
             }));
     },
+
+    reportMetrics(): CachingOfClonesMetrics {
+        return CachingOfClonesMetricsCacheImpl.report();
+    },
 };
 
+/*
+ * Metrics reporting
+ */
+
+const sleepPlease: (timeout: number) => Promise<void> =
+    promisify((a, b) => setTimeout(b, a));
+
+const MetricLogFrequency = 1000 * 60 * 10; // ten minutes
+
+/* This returns a promise that never finishes, unless it throws somehow
+ */
+export function logCachingDirectoryManagerMetricsAllDay(): Promise<void> {
+    return sleepPlease(MetricLogFrequency)
+        .then(() => logger.info(CachingOfClonesMetricsCacheImpl.report().print()))
+        .then(() => logCachingDirectoryManagerMetricsAllDay());
```

Comment:
Is this tail recursion cool stack-wise?

Reply:
the promise returns immediately, it stores the then internally. so it isn't recursive on the same promise thread.

I think so.